### ISSUE TITLE
Search scopes should be grouped along with the items they affect

### ISSUE THREAD

**Issue by jonko0493** [haroohie-club/SerialLoops#273]
Currently most of the search scopes only affect script items, so perhaps they should be grouped with script items? This will require rethinking how we approach the search dialog layout.

**Comment by WiIIiam278** [comment]
Yeah I agree pretty much, the search dialog is a bit weird atm, since most of the search options are useful for script editing.

I think perhaps the best solution to this is to modify how we present search results in this editor, perhaps by introducing a "relevency" scoring for each result and sorting items (ungrouped) by that (with an optional _toggle_ to group items by type)

**Comment by jonko0493** [comment]
Assigning this to you Will -- could redo the search dialog as part of #371.

**Comment by jonko0493** [comment]
Kicking this to v0.4 as we've already done a massive amount of work with the overhaul for v0.3 (and also there's no way I'm gonna be able to get Will to do this before we release 😝)
